Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -67,14 +68,47 @@
         m_lastPoint = beziers[beziersCount - 1].point3;
     }
 
+    STDMETHOD_(void, AddLine)(D2D1_POINT_2F point) {
+        m_geometrySink->AddLine(point);
+        m_lastPoint = point;
+    }
+
+    STDMETHOD_(void, AddBezier)(_In_ CONST D2D1_BEZIER_SEGMENT* bezier) {
+        m_geometrySink->AddBezier(bezier);
+        m_lastPoint = bezier->point3;
+    }
+
+    STDMETHOD_(void, AddQuadraticBezier)(_In_ CONST D2D1_QUADRATIC_BEZIER_SEGMENT* bezier) {
+        m_geometrySink->AddQuadraticBezier(bezier);
+        m_lastPoint = bezier->point2;
+    }
+
+    STDMETHOD_(void, AddQuadraticBeziers)(_In_reads_(beziersCount) CONST D2D1_QUADRATIC_BEZIER_SEGMENT* beziers, UINT32 beziersCount) {
+        m_geometrySink->AddQuadraticBeziers(beziers, beziersCount);
+        m_lastPoint = beziers[beziersCount - 1].point2;
```

Comment:
A quadratic bezier curve has the form, {point1, point2} where point1 is the only control point and point2 is the ending point of the curve.